catalog/imaging/imaging-actionprovider-chip/src/main/webapp/js/chipping.js

```diff
@@ -0,0 +1,154 @@
+function mouseUp() {
+    drag = false;
+    var north =  Math.max(rect.startY + rect.h, rect.startY);
+    var south =  Math.min(rect.startY + rect.h, rect.startY);
+    var east =  Math.max(rect.startX + rect.w,  rect.startX );
+    var west =  Math.min(rect.startX + rect.w,  rect.startX );
+
+    var chipSouth = convertPixelHeightToLat(south);
+    var chipNorth = convertPixelHeightToLat(north);
+    var chipWest = convertPixelWidthToLon(west);
+    var chipEast = convertPixelWidthToLon(east);
+    $('.north').val(chipNorth);
+    $('.south').val(chipSouth);
+    $('.east').val(chipEast);
+    $('.west').val(chipWest);
+}
+
+function draw() {
+    context.beginPath();
+    context.lineWidth="4";
+    context.strokeStyle="#f0e786";
+    context.rect(rect.startX, rect.startY, rect.w, rect.h);
+    context.stroke();
+}
+
+function mouseMove(e) {
+    if (drag) {
+        rect.w = (e.pageX - this.offsetLeft) - rect.startX;
+        rect.h = (e.pageY - this.offsetTop) - rect.startY ;
+        context.drawImage(imageObj, 0, 0, imageObj.width, imageObj.height, 0, 0, canvas.width, canvas.height);
+        draw();
+    }
+}
+
+function drawImage(srcImageUrl) {
+
+    rect = {};
+    imageObj = new Image();
+    imageObj.src = srcImageUrl;
+
+    canvas = document.getElementById('map');
+    context = canvas.getContext('2d');
+
+    imageObj.onload = function() {
+        context.drawImage(imageObj, 0, 0, imageObj.width, imageObj.height, 0, 0, canvas.width, canvas.height);
+    };
+    canvas.addEventListener('mousedown', mouseDown, false);
+    canvas.addEventListener('mouseup', mouseUp, false);
+    canvas.addEventListener('mousemove', mouseMove, false);
+}
+
+function setOnClickListeners() {
+    $('.chip-image').on('click', function() {
+        var wktstring = "POLYGON((";
+        var north = $('.north').val();
+        var south = $('.south').val();
+        var east = $('.east').val();
+        var west = $('.west').val();
+
+        wktstring += west + "%20" + north + ",";
+        wktstring += east + "%20" + north + ",";
+        wktstring += east + "%20" + south + ",";
+        wktstring += west + "%20" + south + ",";
+        wktstring += west + "%20" + north;
+        wktstring += "))";
+        var chipUrl = "/services/catalog/sources/" + source + "/" + id + "?transform=chip&qualifier=overview&location=" + wktstring;
+
+        // Download Image
+        window.location = chipUrl;
+    });
+
+    $('.reset').on('click', function() {
+        drawImage(overviewUrl);
+        $('.north').val("");
+        $('.south').val("");
+        $('.east').val("");
+        $('.west').val("");
+    });
+}
+
+function setBoundariesFromImageGeoLocation(parseLocation) {
+    var array = [];
+    var parsedArray = parseLocation.split(",");
+    $.each(parsedArray, function(index, value) {
+        array.push(parseFloat(value));
+    });
+    north =  array[0];
+    south =  array[1];
+    east =  array[2];
+    west =  array[3];
+}
+
+setUrlParameters();
```

Comment:
Should these function calls go into a [`ready()`](http://api.jquery.com/ready/) handler?